internal/common/customplanmodifier/non_updatable.go

```diff
@@ -0,0 +1,72 @@
+package customplanmodifier
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	planmodifier "github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func NonUpdatableAttributePlanModifier(attribute string) planmodifier.String {
+	return &nonUpdatableAttributePlanModifier{
+		Attribute: attribute,
+	}
+}
+
+type nonUpdatableAttributePlanModifier struct {
+	Attribute string
+}
+
+func (d *nonUpdatableAttributePlanModifier) Description(ctx context.Context) string {
+	return "Ensures that update operations fails when updating an attribute."
+}
+
+func (d *nonUpdatableAttributePlanModifier) MarkdownDescription(ctx context.Context) string {
+	return d.Description(ctx)
+}
+
+func (d *nonUpdatableAttributePlanModifier) PlanModifyString(ctx context.Context, req planmodifier.StringRequest, resp *planmodifier.StringResponse) {
+	var planAttributeValue types.String
+	planFullPath, diag := req.Plan.PathMatches(ctx, GetFullPathExpression(ctx, d.Attribute))
```

Comment:
q: Instead of using req.Plan and req.State, would using req.PlanValue and req.StateValue simplify the implementation?

Reply:
Thanks for the suggestion, using req.PlanValue and req.StateValue makes it a lot simpler.